src/client/Client.cc

```diff
@@ -4689,7 +4685,7 @@ void Client::handle_cap_grant(MetaSession *session, Inode *in, Cap *cap, MClient
         && !_flush(in, new C_Client_FlushComplete(this, in))) {
       // waitin' for flush
     } else if ((old_caps & ~new_caps) & CEPH_CAP_FILE_CACHE) {
-      _release(in);
+      check = _release(in);
```

Comment:
Shouldn't this be "check |= _release(in)" to avoid overriding if check was already true?